libraroo-web/src/bookCatalog/components/CreateBookCatalog.tsx

```diff
@@ -0,0 +1,115 @@
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import bookCatalogService from "../services/bookCatalog.service";
+import { Alert, Box, Button, Container, TextField } from "@mui/material";
+import { isAxiosError } from "axios";
+
+export const CreateBookCatalog: React.FC = () => {
+  const [author, setAuthor] = useState("");
+  const [title, setTitle] = useState("");
+  const [language, setLanguage] = useState("");
+  const [error, setError] = useState<string | null>(null);
+  const [loading, setLoading] = useState(false);
+
+  const navigate = useNavigate(); // React Router's navigation hook
+
+  const handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+    setLoading(true);
+    setError(null); // Clear previous errors
+
+    try {
+      await bookCatalogService.addBookCatalog({
+        author,
+        title,
+        language,
+      });
+
+      // TODO : Navigate to book catalog list screen.
+    } catch (err: any) {
+      if (isAxiosError(err)) {
+        if (err.response) {
+          // Server responded with an error
+          console.log("Server responded:", err.response.status, err.response.data);
+          setError(`Error ${err.response.status}: ${err.response.data.message || "Something went wrong"}`);
+        } else if (err.request) {
+          // Request was made but no response received
+          console.log("No response received:", err.request);
+          setError("Network error: No response from server.");
+        } else {
+          // An error occurred while setting up the request
+          console.log("Error during request setup:", err.message);
+          setError("An unexpected error occurred. Please try again.");
+        }
+      } else {
+        // Handle non-Axios errors
+        console.log("Unexpected error:", err);
+        setError("Unexpected error occurred.");
+      }
```

Comment:
Move logic to a utils func "getApiErrorMessage" also remove irrelevant stuff like status codes